packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.tsx

```diff
@@ -56,17 +85,14 @@ export function QueryWrapper(props: QueryWrapperProps) {
     combineRangeFacetConfig,
   } = props
 
-  // Must store requireConfirmationOnChange in state to break dependency cycle between useImmutableTableQuery and useTableRowsSelection
-  // TODO: consider combining these two hooks, or decoupling them in some other way
-  const [requireConfirmationOnChange, setRequireConfirmationOnChange] =
-    useState(false)
```

Comment:
These changes basically break this dependency cycle, so we can remove this state variable and derive it from the atom.